src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -338,6 +338,79 @@ class profiling : public detail::pimpl<profiling_impl>
   stop() const;
 };
 
+class buffer_impl;
```

Comment:
This is an external API, please carefully document this class e.g. its purpose.  How it is used and why?
Same should be done for other classes in this file, but that can be left for another day.

Reply:
Hi Soren, Added documentation. Please check if it is not clear or need more documentation

Reply:
Hi @stsoe ,
Updated the doc. Please review again.